test/cpp/end2end/xds/xds_end2end_test_lib.cc

```diff
@@ -411,6 +422,13 @@ void XdsEnd2endTest::TearDown() {
     remove(bootstrap_file_);
     gpr_free(bootstrap_file_);
   }
+  // Destruct members so that we can make sure no EventEngine ref is being held
+  backends_.clear();
```

Comment:
This seems potentially fragile, since we may add new data members in the future and not remember to explicitly clear them here.  That would cause the test to hang on shutdown, but it may be hard to understand and debug the problem.

Instead, I suggest using the compiler's guarantee that data members are destroyed in the opposite order in which they are created.  We can add a data member like this as one of the first data members:

```
DefaultEventEngineScope event_engine_;
```

In the ctor, we can initialize this as `event_engine_(CreateEventEngine())`.

Reply:
ofcourse, that's a much easier solution. Stupid brain :D